src/active_item_cache.cpp

```diff
@@ -7,72 +7,75 @@
 #include "debug.h"
 #include "item.h"
 #include "item_stack.h"
+#include "safe_reference.h"
 
-void active_item_cache::remove( item_stack::iterator it, point location )
+void active_item_cache::remove( const item *it )
 {
-    const auto predicate = [&]( const item_reference & active_item ) {
-        return location == active_item.location && active_item.item_iterator == it;
-    };
-    // The iterator is expected to be in this list, as it was added that way in `add`.
-    // But the processing_speed may have changed, so if it's not in the expected container,
-    // remove it from all containers to ensure no stale iterator remains.
-    auto &expected = active_items[it->processing_speed()];
-    const auto iter = std::find_if( expected.begin(), expected.end(), predicate );
-    if( iter != expected.end() ) {
-        expected.erase( iter );
-    } else {
-        for( auto &e : active_items ) {
-            e.second.remove_if( predicate );
-        }
-    }
-    if( active_item_set.erase( &*it ) == 0 ) {
-        // map erase returns number of elements erased
-        debugmsg( "The item isn't there!" );
-    }
+    active_items[it->processing_speed()].remove_if( [it]( const item_reference & active_item ) {
```

Comment:
`remove_if` needs to be followed by `erase` (See [example](https://en.cppreference.com/w/cpp/algorithm/remove))

Reply:
I'm using the `remove_if` member of `std::list` here, not `std::remove_if`, so I'm pretty sure this is right (See [here](https://en.cppreference.com/w/cpp/container/list/remove))